library/azure_rm_cosmosdbaccount.py

```diff
@@ -0,0 +1,488 @@
+                        if ev['default_consistency_level'] == 'eventual':
+                            ev['default_consistency_level'] = 'Eventual'
+                        elif ev['default_consistency_level'] == 'session':
+                            ev['default_consistency_level'] = 'Session'
+                        elif ev['default_consistency_level'] == 'bounded_staleness':
+                            ev['default_consistency_level'] = 'BoundedStaleness'
+                        elif ev['default_consistency_level'] == 'strong':
+                            ev['default_consistency_level'] = 'Strong'
+                        elif ev['default_consistency_level'] == 'consistent_prefix':
+                            ev['default_consistency_level'] = 'ConsistentPrefix'
+                    self.parameters["consistency_policy"] = ev
+                elif key == "locations":
+                    self.parameters["locations"] = kwargs[key]
+                elif key == "database_account_offer_type":
+                    self.parameters["database_account_offer_type"] = kwargs[key]
+                elif key == "ip_range_filter":
+                    self.parameters["ip_range_filter"] = kwargs[key]
+                elif key == "is_virtual_network_filter_enabled":
+                    self.parameters["is_virtual_network_filter_enabled"] = kwargs[key]
+                elif key == "enable_automatic_failover":
+                    self.parameters["enable_automatic_failover"] = kwargs[key]
+                elif key == "capabilities":
+                    self.parameters["capabilities"] = kwargs[key]
+                elif key == "virtual_network_rules":
+                    self.parameters["virtual_network_rules"] = kwargs[key]
+                elif key == "enable_multiple_write_locations":
+                    self.parameters["enable_multiple_write_locations"] = kwargs[key]
+
+        response = None
+
+        self.mgmt_client = self.get_mgmt_svc_client(CosmosDB,
+                                                    base_url=self._cloud_environment.endpoints.resource_manager)
+
+        resource_group = self.get_resource_group(self.resource_group)
+
+        if "location" not in self.parameters:
+            self.parameters["location"] = resource_group.location
+
+        old_response = self.get_databaseaccount()
+
+        if not old_response:
+            self.log("Database Account instance doesn't exist")
+            if self.state == 'absent':
+                self.log("Old instance didn't exist")
+            else:
+                self.to_do = Actions.Create
+        else:
+            self.log("Database Account instance already exists")
+            if self.state == 'absent':
+                self.to_do = Actions.Delete
+            elif self.state == 'present':
+                old_response['locations'] = old_response['failover_policies']
+                if (not compare(self.parameters, old_response, {
```

Comment:
to be honst, the compare method is hard to use. especially to fill in the thrid param.

Reply:
that's the old version..... .I will update :-)

Reply:
already changed